### ISSUE TITLE
Can't resolve 'perf_hooks'

### ISSUE THREAD

**Issue by kautukkundan** [herumi/mcl-wasm#16]
getting `Module not found: Can't resolve 'perf_hooks'` error inside `./node_modules/mcl-wasm/test.js` while trying to use the package. Any fix?

**Comment by herumi** [comment]
`node test` runs well as the following:
https://travis-ci.org/github/herumi/mcl-wasm

https://nodejs.org/api/perf_hooks.html#perf_hooks_performance_now

What command and version did you exec?

**Comment by kautukkundan** [comment]
Hi, 
I am trying to use a module for reactJS which internally uses mcl_wasm. Is this issue because of react?

**Comment by herumi** [comment]
`perf_hooks` is used only for the test, so I don't think that it's a problem.

**Comment by jkawamoto** [comment]
@kautukkundan Are you using a module bundler? Some bundler bundles `test.js`, and it causes the "module not found" error because browsers don't have `perf_hooks`. If you're using webpack, see this comment https://github.com/herumi/mcl-wasm/issues/7#issuecomment-486357004.

**Comment by kautukkundan** [comment]
I did get away with the perf_hooks issue but it seems that the problem is with compatibility of this library with the browser. Now I am getting `mcl.init is not a function`. Is there a way to make this library compatible with the browser? 
what changes could make it work? will changing exports to module.exports work?

**Comment by kautukkundan** [comment]
<img width="345" alt="Screenshot 2020-11-02 at 6 58 31 PM" src="https://user-images.githubusercontent.com/23727056/97873281-69319600-1d3d-11eb-8b8e-708e7121e3a1.png">

doing this in react returns an empty object

![Screenshot 2020-11-02 at 6 58 05 PM](https://user-images.githubusercontent.com/23727056/97873246-59b24d00-1d3d-11eb-91cd-0cbe2ca0704b.png)

**Comment by kautukkundan** [comment]
I made it work! I removed this check
https://github.com/herumi/mcl-wasm/blob/master/mcl.js#L3-L7 
and it now works correctly in react

however it's throwing in new set of errors now
![Screenshot 2020-11-02 at 7 16 23 PM](https://user-images.githubusercontent.com/23727056/97875327-3a68ef00-1d40-11eb-8f0b-1970eb1fdaf8.png)

can anyone help? I am sorry for having so many queries 😅

**Comment by herumi** [comment]
```
mcl-wasm% hexdump -C mcl_c384_256.wasm | head
00000000  00 61 73 6d 01 00 00 00  01 ba 01 1b 60 04 7f 7f  |.asm........`...|
00000010  7f 7f 00 60 03 7f 7f 7f  00 60 02 7f 7f 00 60 01  |...`.....`....`.|
00000020  7f 00 60 01 7f 01 7f 60  03 7f 7f 7f 01 7f 60 04  |..`....`......`.|
00000030  7f 7f 7f 7f 01 7f 60 03  7f 7f 7f 01 7e 60 02 7f  |......`.....~`..|
```

Your error shows `3c 21 44 4f`.
What file is this?

**Comment by herumi** [comment]
@kautukkundan
Could you try the latest dev branch https://github.com/herumi/mcl-wasm/tree/dev ?
I have to refactor some code yet, but it seems to run well.

```
import React from 'react';
import ReactDOM from 'react-dom';
import mcl from 'mcl-wasm';

class Layout extends React.Component {
  constructor() {
    super();
    this.state = { msg:"initializing..." }
    mcl.init().then(() => {
      const x = new mcl.Fr()
      x.setStr('123')
      this.setState({
        msg: x.getStr()
      })
    })
  }
  render() {
    return (
      <h2>{this.state.msg}</h2>
    );
  }
}

const app = document.getElementById('app');
ReactDOM.render(<Layout />, app);
```

**Comment by kautukkundan** [comment]
I had forked and edited the lib to make it work temporarily. The issue was loading the .wasm file with the fetch API. So I moved the .wasm file to public folder and it worked. 
https://github.com/kautukkundan/mcl-wasm

I'll surely checkout the new changes in this repo and let you know. 
Thank you so much

**Comment by herumi** [comment]
I make a tiny sample of React. 

https://github.com/herumi/misc/tree/master/react/test2

To import mcl has been changed.

https://github.com/herumi/misc/blob/master/react/test2/src/index.js#L3
```
import React from "react";
import ReactDOM from "react-dom";
import mcl from "mcl-wasm/browser/src/index-browser.js";

console.log('AAA')

class Layout extends React.Component {
  constructor () {
    super()
    this.state = { msg:"initializing..." }
    console.log('BBB')
    console.log(mcl)
    mcl.init().then(() => {
      const x = new mcl.Fr()
      x.setStr('123456')
      this.setState({
        msg: x.getStr()
      })
    })
  }
  render () {
    return <h2>{this.state.msg}</h2>
  }
}

const app = document.getElementById('app');
ReactDOM.render(<Layout/>, app);
```

**Comment by herumi** [comment]
I fixed the path at the version v0.7.1.

You can import `mcl-wasm` in React by
```
import mcl from "mcl-wasm/browser";
```
